ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java

```diff
@@ -290,16 +314,68 @@ public Object handle(Request request, Response response) throws Exception {
       arguments = cswTransformArgumentsAdapter();
     }
 
-    attachFileToResponse(request, response, queryResponseTransformer, combinedResponse, arguments);
+    String warning = null;
+    if (resultsNotToExport.size() > 0) {
+      List<String> requiredAttrList = new ArrayList<>();
+      String idName = mapAliasAttributes(queryResponseTransformer, requiredAttrList, arguments);
+      if (results.size() == 0) {
+        LOGGER.debug("0 Results to export due to missing required field(s): {}", requiredAttrList);
+        response.status(HttpStatus.BAD_REQUEST_400);
+        return ImmutableMap.of(
+            "message", String.format("Result(s) missing required field(s): %s", requiredAttrList));
+      }
+      warning =
+          getWarningMessage(
+              String.format(
+                  "Following not exported, missing required field(s): %s", requiredAttrList),
+              idName,
+              resultsNotToExport);
+    }
+
+    attachFileToResponse(
+        request, response, queryResponseTransformer, combinedResponse, arguments, warning);
 
     return "";
   }
 
+  private String getWarningMessage(
+      String warningMsg, String idName, List<String> resultsNotToExport) {
+    int count = 0;
+    for (String resultId : resultsNotToExport) {
+      warningMsg += String.format("\\n%s) %s: %s", ++count, idName, resultId);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.